plugin/src/main/kotlin/nl/avisi/anowalke/toolWindow/FluffyListener.kt

```diff
@@ -42,42 +47,89 @@ class FluffyListener(val project: Project) : ToolWindowManagerListener {
+                log.warn("Extract function retrieved ${expressions.size} expressions.")
+                expressions = sanitiser.sanitise(expressions)
+            }
+            log.warn("Sanitise function retrieved ${expressions?.size} expressions.")
+            return expressions?.map { expression ->
+                Identifier(file.name, expression)
+            }
+        } else if (file.isDirectory) {
+            log.warn("File: ${file.name} is a directory. Check children for java files.")
+            return file.children.flatMap { getJavaFiles(it).orEmpty() }
         }
         return null
     }
 
     private fun getFilesAsIdentifier(file: VirtualFile): List<Identifier>? {
-        if (file.isDirectory) {
-            // Zoek naar files in de java map ipv files met filetype java, omdat hij sommige filetypes door elkaar haalt
-            if (file.name == "java") {
-                return getJavaFiles(file)?.map { expression ->
-                    Identifier(file.name, expression)
-                }
-            }
+        log.warn("Initiate getFilesAsIdentifier function.")
+        // Zoek naar files in de java map ipv files met filetype java, omdat hij sommige filetypes door elkaar haalt
+        if (file.isDirectory && file.name == "java") {
+            log.warn("Current file: '${file.name}' is a java directory.")
+            val identifiers = getJavaFiles(file)
+            log.warn("getJavaFiles function retrieved ${identifiers?.size} identifiers.")
+            return identifiers
+        } else if (file.isDirectory && file.name != "java") {
+            log.warn("Current file: ${file.name} is not a java directory. Check Children for a directory named 'java'.")
+            return file.children.flatMap { getFilesAsIdentifier(it).orEmpty() }
         } else {
-            file.children.forEach { getFilesAsIdentifier(it) }
+            log.warn("Current file: ${file.name} is not a directory. Ignore file.")
         }
         return null
     }
 
-    private fun scanProject() {
+    private fun scanProject(): List<Identifier> {
+        log.warn("Initiate scanProject function.")
         val identifiers = ProjectRootManager.getInstance(project).contentSourceRoots.flatMap {
             getFilesAsIdentifier(it).orEmpty()
         }
+        identifiers.forEach { log.warn("Expression: ${it.expression}") }.also { log.warn("Aloha") }
+        return identifiers
+    }
+
+    private fun findShotgunSurgery(identifiers: List<Identifier>): Map<String, List<Identifier>> {
+        // Kijk naar overeenkomsten in expressies
+        // vereiste 1: Wanneer een expressie overeenkomt met een classname of een deel van een classname
+        // (kijk voor de gehele class name als het maar met een deel overeenkomt)
+        // en de expressie daarna een hit oplevert negeer die dan.
+        // vereiste 2: Iets met initialisatie, voor nu niet meenemen.
+
+        val testIdentifiers = listOf(
+            Identifier("dier", "koe"),
```

Comment:
Dit moet uiteindelijk nog iets gaan doen met de api?

Reply:
Ja... Omdat de testinput die ik teruggeef vanuit de api voor iedere class hetzelfde teruggeeft was letterlijk ieder woord in de array een 'hit' dus daarom had ik er hier nog een extra iets bijgetoverd om in ieder geval te kunnen kijken of het nu deed wat het hoorde te doen. Kan weg als ik gewoon een unittest geschreven heb.

Reply:
Dan snel een unittest schrijven :)